sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/UnsafeRowUtils.scala

```diff
@@ -204,8 +204,8 @@ object UnsafeRowUtils {
    * e.g. this is not true for non-binary collations (any case/accent insensitive collation
    * can lead to rows being semantically equal even though their binary representations differ).
    */
-  def isBinaryStable(dataType: DataType): Boolean = dataType.existsRecursively {
-    case st: StringType => CollationFactory.fetchCollation(st.collationId).isBinaryCollation
-    case _ => true
+  def isBinaryStable(dataType: DataType): Boolean = !dataType.existsRecursively {
```

Comment:
why `isBinaryStable` is in `UnsafeRowUtils`. Is the implementation bound somehow to unsafe row?

Why it is not in `DataTypeUtils`, for example?

Reply:
Not sure what are the consequences of moving this function, do you know if we can do that @dbatomic?